metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -302,20 +303,39 @@ func (sink *StackdriverSink) requestSender(reqQueue chan *sd_api.CreateTimeSerie
 	}
 }
 
+func marshalRequestAndLog(printer func([]byte), req *sd_api.CreateTimeSeriesRequest) {
+	reqJson, errJson := json.Marshal(req)
+	if errJson != nil {
+		printer(reqJson)
+	} else {
+		glog.Errorf("Couldn't marshal Stackdriver request %v", errJson)
+	}
+}
+
 func (sink *StackdriverSink) sendOneRequest(req *sd_api.CreateTimeSeriesRequest) {
 	startTime := time.Now()
 	empty, err := sink.stackdriverClient.Projects.TimeSeries.Create(fullProjectName(sink.project), req).Do()
 
 	var responseCode int
 	if err != nil {
 		glog.Warningf("Error while sending request to Stackdriver %v", err)
+		if glog.V(2) {
```

Comment:
How about adding a comment explaining this if?

Reply:
Added.